gateway/api.go

```diff
@@ -1075,27 +1134,26 @@ func (gw *Gateway) polHandler(w http.ResponseWriter, r *http.Request) {
 
 func (gw *Gateway) apiHandler(w http.ResponseWriter, r *http.Request) {
 	apiID := mux.Vars(r)["apiID"]
-	oasTyped := r.FormValue("type") == "oas"
 
 	var obj interface{}
 	var code int
 
 	switch r.Method {
 	case "GET":
 		if apiID != "" {
-			log.Debug("Requesting API definition for", apiID)
-			obj, code = gw.handleGetAPI(apiID, oasTyped)
+			log.Debugf("Requesting API definition for %q", apiID)
+			obj, code = gw.handleGetAPI(apiID, false)
 		} else {
 			log.Debug("Requesting API list")
 			obj, code = gw.handleGetAPIList()
 		}
 	case "POST":
 		log.Debug("Creating new definition file")
-		obj, code = gw.handleAddOrUpdateApi(apiID, r, afero.NewOsFs(), oasTyped)
+		obj, code = gw.handleAddOrUpdateApi(apiID, r, afero.NewOsFs(), false)
 	case "PUT":
 		if apiID != "" {
-			log.Debug("Updating existing API: ", apiID)
-			obj, code = gw.handleAddOrUpdateApi(apiID, r, afero.NewOsFs(), oasTyped)
+			log.Debugf("Updating existing API: %q", apiID)
+			obj, code = gw.handleAddOrUpdateApi(apiID, r, afero.NewOsFs(), false)
```

Comment:
Ideally, as you're reading *Request for apiID in this function, create a struct with all the possible arguments as fields. Don't pass along *Request, pass *apiMethodParams; this is a context switch between the http layer, and things that read request should stay here.

Can we pass aerofs.NewOsFs instead of invoking it? Is the returned value an interface that we can mock in tests? By invoking it, you allocate the fs too early, before possible error returns.